roles/vault_agent/tasks/main.yml

```diff
@@ -0,0 +1,95 @@
+- name: Download Vault Hashicorp gpg key
+  ansible.builtin.get_url:
+    url: https://apt.releases.hashicorp.com/gpg
+    dest: /usr/share/keyrings/hashicorp-archive-keyring.asc
+    mode: "0644"
+    owner: root
+    checksum: sha256:cafb01beac341bf2a9ba89793e6dd2468110291adfbb6c62ed11a0cde6c09029
+
+- name: Add Vault repository
+  ansible.builtin.apt_repository:
+    repo: "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.asc] https://apt.releases.hashicorp.com {{ ansible_distribution_release }} main"
+    state: present
+
+- name: Intall Vault
+  ansible.builtin.apt:
+    update_cache: true
+    name:
+      - vault
+
+- name: Retrieve role_id and secret_id from Vault
+  community.hashi_vault.vault_kv2_get:
+    path: certificat-web-id
+  register: certificat_secrets
+  run_once: true
+  delegate_to: localhost
+  become: false
+
+- name: Create vault agent workdir
+  ansible.builtin.file:
+    path: /root/vault_agent_certificat
+    state: directory
+    mode: '0755'
+    owner: root
+    group: root
+
+- name: Create role_id file
+  vars:
+    content: "{{ certificat_secrets.secret.role_id }}"
+  ansible.builtin.template:
+    src: content.j2
+    dest: /root/vault_agent_certificat/role_id
+    mode: '0600'
+    owner: root
+    group: root
+
+- name: Create secret_id file
+  vars:
+    content: "{{ certificat_secrets.secret.secret_id }}"
+  ansible.builtin.template:
+    src: content.j2
+    dest: /root/vault_agent_certificat/secret_id
+    mode: '0600'
+    owner: root
+    group: root
+
+- name: Copy agent-config.hcl
+  ansible.builtin.template:
+    src: agent-config.hcl.j2
+    dest: /root/vault_agent_certificat/agent-config.hcl
+    mode: '0644'
+    owner: root
+    group: root
+  notify: Restart vault-agent-certificates
+
+- name: Copy retrieving_cert.tmpl
+  ansible.builtin.template:
+    src: retrieving_cert.tmpl.j2
+    dest: /root/vault_agent_certificat/retrieving_cert.tmpl
```

Comment:
It may be a good idea to change the template variable delimiters for this task so we don't have to escape all the `{{` and `}}` in the `retrieving_cert.tmpl` file. This can be achieved by adding the following line : `variable_start_string: <<` (see [variable_start_string](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/template_module.html#parameter-variable_start_string))